iothub/device/src/Common/Security/CertificateInstaller.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    internal static class CertificateInstaller
+    {
+        private static readonly object s_certOperationsLock = new object();
+
+        /// <summary>
+        /// Ensures the specified certs (presumably in a chain) are installed in the cert store.
+        /// </summary>
+        /// <remarks>
+        /// Because Intermediate Authorities may have been issued by the uploaded CA, the application must present the full chain of
+        /// certificates from the one used during authentication to the one uploaded to the service.
+        /// See <see href="TODO (Sample ref) - Sindhu"/>
```

Comment:
nit: we might want to remove this before checking it in :)

Reply:
I thought it was ok to leave it and update it after my next PR. Not sure if it is ok to leave it or not. What does everyone follow?

Reply:
I would say if your next PR follows this one immediately, it's ok to leave it here.